### PR TITLE
Fix Celery Cache Warmup

### PR DESCRIPTION
**Pull request by AmberCharitos**
Resolves this bug: https://warthogs.atlassian.net/browse/CSS-7244

Please note that the celery worker needs the host of the server application which depends on the name given when deploying the application. This information should be transferred in a relation between the server and celery charm but at the moment no such relation exists. I have created a Jira ticket for separating the server, worker and beat charms and providing relations between them: https://warthogs.atlassian.net/browse/CSS-7260. But in order to resolve this error in the meantime I have made this a customizable config.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -189,3 +189,7 @@ options:
     description: Indicates whether or not event parameters sent to Sentry should be redacted.
     default: false
     type: boolean
+  server-alias:
+    description: The alias the server charm has been deployed with if it differs from the default.
```
[the remark this answers is not in the document]
Comment (AmberCharitos) [reply config.yaml:193]: Yeah unfortunately - if you have alternative ways of doing this in the short term I would love to know. In the long term as I mentioned above I think this should be via a relation between the 2 charms.

**Review by gtato — APPROVED** [review]
As @AmberCharitos said, there is no easy way to fix this quickly, until we have relations between charms, so this is ok for me.

File: src/charm.py [hunk 2]
```diff
@@ -353,6 +355,10 @@ def _update(self, event):
                     }
                 },
             )
+
+            # Open port for cache warm-up.
+            self.model.unit.open_port(port=APPLICATION_PORT, protocol="tcp")
```
Comment (kelkawi-a) [inline src/charm.py:360]: Should this service port only be opened for a specific `charm-function`?

**Review by kelkawi-a — APPROVED** [review]
LGTM

File: src/charm.py (hunk 2 above)
Comment (AmberCharitos) [reply src/charm.py:360]: Yes, only for the `app-gunicorn`

**Pull request merged**